### PR TITLE
Handle exceeding browser dimensions

### PR DESCRIPTION
**Pull request by j0weiss**
Based on the issue #123 I added two things:
* the agent also sends the resolution when running on macOS and
* `ffmpeg` gets updated width and height based on the actual dimensions of the host i.e. if the given dimensions exceed the resolution of the system the actual dimensions will be used.

### CODE REVIEW

File: internal/desktop_browser.py [hunk 1]
```diff
@@ -365,6 +367,19 @@ def on_start_recording(self, task):
                     self.execute_js(SET_ORANGE)
                     time.sleep(1)
                 task['video_file'] = os.path.join(task['dir'], task['prefix']) + '_video.mp4'
+                if platform.system() == 'Windows':
+                    from win32api import GetSystemMetrics
+                    self.screen_width = GetSystemMetrics(0)
+                    self.screen_height = GetSystemMetrics(1)
+                elif platform.system() == 'Darwin':
+                    from AppKit import NSScreen
```
Comment (pmeenan) [inline internal/desktop_browser.py]: Can you put a try/except around this just in case it's not installed?

File: internal/desktop_browser.py [hunk 2]
```diff
@@ -56,6 +56,8 @@ def __init__(self, path, options, job):
         self.task = None
         self.cpu_start = None
         self.throttling_cpu = False
+        self.screen_width = None
```
Comment (pmeenan) [inline internal/desktop_browser.py]: Please set the defaults to 1920/1200 here so None can never happen

File: internal/desktop_browser.py [hunk 3]
```diff
@@ -365,6 +367,19 @@ def on_start_recording(self, task):
...
                 task['video_file'] = os.path.join(task['dir'], task['prefix']) + '_video.mp4'
+                if platform.system() == 'Windows':
+                    from win32api import GetSystemMetrics
+                    self.screen_width = GetSystemMetrics(0)
+                    self.screen_height = GetSystemMetrics(1)
+                elif platform.system() == 'Darwin':
+                    from AppKit import NSScreen
+                    self.screen_width = int(NSScreen.screens()[0].frame().size.width)
+                    self.screen_height = int(NSScreen.screens()[0].frame().size.height)
+                else:
```
Comment (pmeenan) [inline internal/desktop_browser.py]: No need for the else clause if the defaults are set above.

File: internal/webpagetest.py [hunk 4]
```diff
@@ -81,6 +81,10 @@ def __init__(self, options, workdir):
                 from win32api import GetSystemMetrics
                 self.screen_width = GetSystemMetrics(0)
                 self.screen_height = GetSystemMetrics(1)
+            elif platform.system() == 'Darwin':
```
Comment (pmeenan) [inline internal/webpagetest.py]: try/except block here as well please.

File: internal/desktop_browser.py [hunk 5]
```diff
@@ -365,6 +367,19 @@ def on_start_recording(self, task):
...
+                    self.screen_width = 1920
+                    self.screen_height = 1200
+                task['width'] = self.screen_width if task['width'] > self.screen_width else task['width']
```
Comment (pmeenan) [inline internal/desktop_browser.py]: A bit easier to read if this is changed to min(self.screen_width, task['width']) (and for height).

**Comment by j0weiss** [comment]
I added some changes according to your hints.
Thanks for the feedback.

**Comment by pmeenan** [comment]
Looks good, thanks for the patch.

**Pull request merged**